src/http2_stream.erl

```diff
@@ -574,10 +580,17 @@ check_content_length(Stream) ->
         undefined ->
             ok;
         _Other ->
-            case Stream#stream_state.request_body_size =:= binary_to_integer(ContentLength) of
-                true ->
-                    ok;
-                false ->
+            try binary_to_integer(ContentLength) of
+                Integer ->
+                    case Stream#stream_state.request_body_size =:= Integer of
+                        true ->
+                            ok;
+                        false ->
+                            rst_stream_(?PROTOCOL_ERROR, Stream),
+                            rst_stream
+                    end
+            catch
+                _:_ ->
```

Comment:
I don't think it will. As I understand it, the try/catch only applies to the function call before `of`.

To test this, I inserted a `throw(explode)` before the call to `rst_stream_/2` at line 589. If the catch was protecting that, the test would fail on not getting a rst_stream frame as a reply, but that's not what happened.

What happened is this 

```
08:14:06.282 [debug] terminate reason: {bad_return_value,explode}
```